libyara/exception.h

```diff
@@ -57,7 +57,27 @@ static LONG CALLBACK exception_handler(
   return EXCEPTION_CONTINUE_SEARCH;
 }
 
-#define YR_TRYCATCH(_do_,_try_clause_, _catch_clause_)                  \
+#ifdef _MSC_VER
+
+#include <excpt.h>
+
+#define YR_TRYCATCH(_do_,_try_clause_,_catch_clause_)           \
+  do                                                            \
+  {                                                             \
+    if (_do_)                                                   \
+    {                                                           \
+      __try                                                     \
+      { _try_clause_ }                                          \
+      __except(exception_handler(GetExceptionInformation))      \
```

Comment:
I believe this needs to be "GetExceptionInformation()" since GetExceptionInformation is a function that you need to invoke in order to get the _EXCEPTION_POINTERS*

After making that change I can confirm the code compiles on VS2015.  Will look at testing the actual exception handling later today.